cirq/circuits/qasm_output.py

```diff
@@ -110,8 +110,8 @@ def _decompose_(self, qubits: Sequence[ops.QubitId]) -> ops.OP_TREE:
         b = self.y * -2 / np.pi + 0.5
         c = self.z * -2 / np.pi + 0.5
 
-        yield self.before0(q0)
-        yield self.before1(q1)
+        yield self.before1(q0)
```

Comment:
The 1 matched with 0 is a bit distressing. We must have an even number of 0-1 swaps on this path somehow.

Reply:
Yes, quite distressing. What's the problem with switching the roles of `before0` and `before1` so that 0 matches with 0 and 1 with 1 here?

Reply:
Fixed by storing the kak decomposition directly instead of splitting it apart